test/integration/test_context_based_restrictions_v1.py

```diff
@@ -0,0 +1,564 @@
+        }
+
+        # Construct a dict representation of a Resource model
+        resource_model = {
+            'attributes': [account_id_resource_attribute_model, service_name_resource_attribute_model],
+        }
+
+        create_rule_response = self.context_based_restrictions_service.create_rule(
+            contexts=[rule_context_model],
+            resources=[resource_model],
+            description='SDK TEST - this is an example of rule',
+            transaction_id=self.getTransactionID()
+        )
+
+        assert create_rule_response.get_status_code() == 201
+        rule = create_rule_response.get_result()
+        assert rule is not None
+        TestContextBasedRestrictionsV1.rule_id = rule['id']
+
+    @needscredentials
+    def test_create_rule_with_service_not_cbr_enabled_error(self):
+        # Construct a dict representation of a RuleContextAttribute model
+        rule_context_attribute_model = {
+            'name': 'networkZoneId',
+            'value': TestContextBasedRestrictionsV1.zone_id,
+        }
+
+        # Construct a dict representation of a RuleContext model
+        rule_context_model = {
+            'attributes': [rule_context_attribute_model],
+        }
+
+        # Construct a dict representation of a ResourceAttribute model
+        account_id_resource_attribute_model = {
+            'name': 'accountId',
+            'value': TestContextBasedRestrictionsV1.test_account_id,
+        }
+
+        service_name_resource_attribute_model = {
+            'name': 'serviceName',
+            'value': "cbr-not-enabled",
+        }
+
+        # Construct a dict representation of a Resource model
+        resource_model = {
+            'attributes': [account_id_resource_attribute_model, service_name_resource_attribute_model],
+        }
+
+        with pytest.raises(ApiException, match="400"):
+            self.context_based_restrictions_service.create_rule(
+                contexts=[rule_context_model],
+                resources=[resource_model],
+                description='SDK TEST - this is an example of rule',
+                transaction_id=self.getTransactionID()
+            )
+
+
+    @needscredentials
+    def test_list_rules(self):
+        list_rules_response = self.context_based_restrictions_service.list_rules(
+            account_id=TestContextBasedRestrictionsV1.test_account_id,
+            transaction_id=self.getTransactionID(),
+        )
+
+        assert list_rules_response.get_status_code() == 200
+        rule_list = list_rules_response.get_result()
+        assert rule_list is not None
+
+    @needscredentials
+    def list_rules_with_missing_required_account_id_parameter_error(self):
+        with pytest.raises(ApiException, match="404"):
+            self.context_based_restrictions_service.list_rules(
+                transaction_id=self.getTransactionID(),
+            )
+
+    @needscredentials
+    def list_rules_with_invalid_account_id_parameter_error(self):
```

Comment:
The other 3 SDKs expect `400` status code in this case. Can I ask you to double check whether `404` or `400` needed here as expected status code?

Reply:
I re ran this test and 400 is the expected status code. Thank you for catching this!